modules/autotagging-commons/common/src/main/java/org/opensearch/rule/storage/AttributeValueStore.java

```diff
@@ -40,11 +39,21 @@ default void remove(K key, V value) {
     void remove(K key);
 
     /**
-     * Returns the values associated with the key
-     * @param key in the data structure
+     * Returns the values associated with the given key, including prefix matches,
+     * along with their match scores.
+     * For example, searching for "str" may also return results for "st" and "s".
+     * @param key the key to look up
+     */
+    default Map<V, Float> getWithMatchScore(K key) {
+        return new HashMap<>();
+    }
+
+    /**
+     * Returns the values that exactly match the given key.
+     * @param key the key to look up
      */
-    default List<Set<V>> getAll(K key) {
-        return new ArrayList<>();
+    default Set<V> getExactMatch(K key) {
+        return new HashSet<>();
```

Comment:
Can this we generalised and maybe leverage the above method to filter matches with a score of 100% match